server/consumer.go

```diff
@@ -68,6 +65,17 @@ type ConsumerConfig struct {
 	FlowControl     bool          `json:"flow_control,omitempty"`
 	HeadersOnly     bool          `json:"headers_only,omitempty"`
 
+	// Pull based options.
+	MaxRequestBatch   int           `json:"max_batch,omitempty"`
+	MaxRequestExpires time.Duration `json:"max_expires,omitempty"`
```

Comment:
what are your thoughts about defaults for these? and maximums? Definitely think these need sane defaults, eventhough clients are supposed to set defaults here and not allow uncapped I think its a bad default to allow uncapped as per recent mails

Reply:
If the request still remains it is legit, so not sure about default limits.

The other issue of fairness is still a consideration.

Reply:
OK, well we can park it till fairness is clearer I suppose as it relates to that ultimately